### PR TITLE
added url validation, fixed update uniqueness

### PR DESCRIPTION
**Pull request by Dkrisztan**

### CODE REVIEW

File: packages/backend/src/links/links.service.ts [hunk 1]
```diff
@@ -1,45 +1,64 @@
...
 @Injectable()
 export class LinksService {
   constructor(private readonly prisma: PrismaService) {}
   async create(createLinkDto: CreateLinkDto) {
     try {
-      return await this.prisma.link.create({ data: createLinkDto });
+      if (this.checkUrl(createLinkDto.url)) {
+        return await this.prisma.link.create({ data: createLinkDto })
+      } else {
+        throw new HttpException('The url you entered is not found!', 404)
```
Comment (Tschonti) [inline packages/backend/src/links/links.service.ts]: I think this should be a BadRequest as well. We usually send not found when the primary entity of the controller is not found, so link in this case.

File: packages/backend/src/links/links.service.ts [hunk 2]
```diff
@@ -1,45 +1,64 @@
...
+    try {
+      return await this.prisma.link.delete({ where: { id } })
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === 'P2002') {
+          throw new BadRequestException('This slug violates the unique constraint!')
+        }
+      }
+    }
+  }
```
Comment (Tschonti) [inline packages/backend/src/links/links.service.ts]: This was meant for the update method, right? No point of checking this on the delete method

File: packages/backend/src/links/links.service.ts [hunk 3]
```diff
@@ -1,45 +1,64 @@
...
+
+  async checkUrl(url: string) {
+    const response = await fetch(url)
+    if (response.statusText === 'OK') {
+      return true
+    } else {
+      return false
```
Comment (Tschonti) [inline packages/backend/src/links/links.service.ts]: `return response.statusText === 'OK'` is much more compact

**Review by Tschonti — CHANGES_REQUESTED** [review]
On top of the changes I (we) suggested on slack (axios, head request), a few more down here:

File: packages/backend/src/links/links.service.ts (hunk 2 above)
Comment (Dkrisztan) [reply packages/backend/src/links/links.service.ts]: Yes! I copy pasted to the wrong place, my bad. Will fix

**Review by Tschonti — CHANGES_REQUESTED** [review]
Nice 👌 
Just one more thing left: the link should be validated in the update method as well

**Comment by Tschonti** [comment]
fixed in other branch/PR

**Pull request closed without merging**